### PR TITLE
Refactor to use async-nats

### LINKED ISSUE #283: [BUG] Can't connect to `connect.ngs.global` using wash ctl
**Describe the bug**
When attempting to use an NGS credsfile to connect to connect.ngs.global, I get the following warning and error:
```
[2022-06-08T11:41:26Z WARN  rustls::conn] Sending fatal alert BadCertificate

presented server name type wasn't supported
```
**To Reproduce**
1. Get a set of credentials for `connect.ngs.global`
1. Run `wash ctl get hosts -r connect.ngs.global --ctl-credsfile <path/to/credsfile>`
1. See error

**Expected behavior**
I would expect, if given valid credentials, that we could connect through NGS just fine

**Additional context**
I've also been able to reproduce this error with a combination of `--ctl-seed` and `--ctl-jwt`

### LINKED ISSUE #284: [BUG] Occasional hang on `ctl` commands
**Describe the bug**
On `v0.11.0`, I've found that occasionally my commands will go far beyond the timeout value for ctl commands. For example:
```
 on ☁️  dev (us-east-2) took 2s  ~/demo/leafnode ➜
wash ctl start actor wasmcloud.azurecr.io/clinicapi:0.2.3 --host-id NBWURVUSDQ7I235EKRAEYSHRYCITOQBOCA573SWJODAHFWRAXUGYL4WH
>>> ⢋⠀  Starting actor wasmcloud.azurecr.io/clinicapi:0.2.3 ...
^C

 on ☁️  dev (us-east-2) took 55s  ~/demo/leafnode ➜
wash ctl start actor wasmcloud.azurecr.io/clinicapi:0.2.3 --host-id NBWURVUSDQ7I235EKRAEYSHRYCITOQBOCA573SWJODAHFWRAXUGYL4WH

Actor wasmcloud.azurecr.io/clinicapi:0.2.3 started on host NBWURVUSDQ7I235EKRAEYSHRYCITOQBOCA573SWJODAHFWRAXUGYL4WH
```

As you can see in the example above, the start actor command ran for 55 seconds before I CTRL+C'ed it. Our timeouts for starting an actor, when a host ID is supplied, is around 3000ms so it shoudn't take this long to time out.

More information is needed in terms of identifying the place this is timing out, it could potentially be on the initial NATS connect which might be fixed by moving to the `async-nats` crate

### PR DESCRIPTION
**Pull request by brooksmtownsend**
This PR fixes #283 
I ran a `ctl get hosts` 1000 times and could not reproduce #284, so I'm saying this fixes #284 since I could reproduce it consistently (@connorsmith256 would be curious if you can give this a try too)

This is a breaking change because:
- It removes the ability to authenticate with a NATS server with only an nkey seed
- I removed the `--ack-timeout-ms` flag in favor of a ubiquitous `--timeout-ms` for the control interface.
  - In the case where you don't provide a different timeout for starting actors/providers, I automatically increase the timeout if you don't add `--skip-wait` to avoid automatically "failing" to start a provider if it's your first download
  
This PR also fixes a bug where the default auction timeout was used for `wash ctl get hosts`, ignoring the specified timeout.

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -227,35 +227,32 @@ pub(crate) async fn nats_client_from_opts(
...
-        anats::Options::with_jwt(
-            move || Ok(jwt_contents.clone()),
-            move |nonce| kp.sign(nonce).unwrap(),
-        )
-        .connect(&nats_url)
+        async_nats::ConnectOptions::with_jwt(jwt_contents, move |nonce| {
+            let key_pair = kp.clone();
+            async move { key_pair.sign(&nonce).map_err(async_nats::AuthError::new) }
+        })
+        .connect("localhost")
```
Comment (connorsmith256) [inline src/util.rs]: Should this still be `&nats_url`?

**Review by connorsmith256 — COMMENTED** [review]
Someone more familiar with this code should approve, but FWIW I can no longer reproduce https://github.com/wasmCloud/wash/issues/284 after compiling from this branch 👍

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -58,7 +58,7 @@ jobs:
           release_name: Release ${{ github.ref }}
           token: ${{ secrets.GITHUB_TOKEN }}
           draft: false
-          prerelease: true
+          prerelease: false
```
Comment (ghost) [inline .github/workflows/release.yml]: Could this be in a separate PR? I know it fixes #286 but might look confusing when looking back through commit history

File: Cargo.toml [hunk 3]
```diff
@@ -52,16 +53,15 @@ thiserror = "1.0"
...
-
-wasmcloud-test-util = "0.3.3"
-weld-codegen = "0.4.5"
 which = "4.2.2"
-crossbeam-channel = "0.5.2"
 cloudevents-sdk = "0.4.0"
 
+wascap = "0.8.0"
+weld-codegen = "0.4.6"
+wasmcloud-control-interface = "0.16.2"
```
Comment (ghost) [inline Cargo.toml:61]: While i'm here - could we set up Dependabot to handle creating version upgrade PRs for us?

**Review by ghost — DISMISSED** [review]

File: src/util.rs (hunk 1 above)
Comment (brooksmtownsend) [reply src/util.rs]: Good catch! I think this was a documentation copy fail (cursed jwt nonce closure async blah)

File: Cargo.toml (hunk 3 above)
Comment (brooksmtownsend) [reply Cargo.toml:61]: I wouldn't mind doing that, just in a separate PR 😄 I'll file an issue real quick to track that

Comment (ghost) [reply Cargo.toml:61]: Yep sounds good!

File: .github/workflows/release.yml (hunk 2 above)
Comment (brooksmtownsend) [reply .github/workflows/release.yml]: Good call 👍🏻

Comment (brooksmtownsend) [reply .github/workflows/release.yml]: Dropped that commit

**Review by ghost — APPROVED** [review]
